### ISSUE TITLE
Pretrain pegasus

### ISSUE THREAD

**Issue by PingYu-iris** [google-research/pegasus#58]
Hi there, 

After reading your paper, I am a little confused about pretrain procedure of Pegasus. 

As shown in Figure 1 in the paper, given a document pegasus selected target text from the document and let the rest of text be input text. 

1. pegasus model masked several tokens in input and let encoder to predict masked tokens as MLM loss. However, the paper stated that the pegasus_large model deleted the MLM loss part. Does that mean the final model takes masked inputs to the encoder and decoder leveraged shifted target text and encoder outputs to generate target text only using the next token prediction loss? **If so, what is the difference between the pegasus pretrained model and BART model** (BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension). The BART model also takes noised text input to their encoder and uses the next token prediction loss for decoder outputs. 

2. **What is the strategy for selecting target sentences as shown in Fig.1**? Is that the Gap Sentence Generation procedure?

Thanks

**Comment by JingqingZ** [comment]
> 1. pegasus model masked several tokens in input and let encoder to predict masked tokens as MLM loss. However, the paper stated that the pegasus_large model deleted the MLM loss part. Does that mean the final model takes masked inputs to the encoder and decoder leveraged shifted target text and encoder outputs to generate target text only using the next token prediction loss? **If so, what is the difference between the pegasus pretrained model and BART model** (BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension). The BART model also takes noised text input to their encoder and uses the next token prediction loss for decoder outputs.

Yes, in the final model, only GSG is applied and there is no MLM.

Document: ABCDE
* BART: input: A_C_E, target: ABCDE
* PEGASUS: input: A_C_E, target: BD

> 2. **What is the strategy for selecting target sentences as shown in Fig.1**? Is that the Gap Sentence Generation procedure?

Fig 1 demonstrates both GSG and MLM. There is no specific sentence selection strategy for Fig 1 as the figure is only for demo purpose. Any strategy (e.g. random, lead or principal) would fit the procedure shown in Fig 1.

Hope this may answer your questions. Thanks!

**Comment by PingYu-iris** [comment]
<img width="451" alt="Screen Shot 2020-07-14 at 1 05 49 PM" src="https://user-images.githubusercontent.com/23408859/87454975-bf35ce80-c5d2-11ea-9860-49df8d12975d.png">

1.  I do not understand " PEGASUS: input: A_C_E, target: BD". 
 As you deleted MLM loss, if the input for the decoder is " <bos>ABCDE" then the target would be "ABCDE<eos>". Then what is your "target: BD"?

2. In fig1, I don't understand how to select input text for encoder and decoder?
In this example, your text is "Pegasus if mythical. It is pure white. It names the model." 
The input for the encoder is masked "Pegasus if mythical. It names the model."
The input for the decoder is "<bos> It is pure white."
The target for the decoder is "It is pure while. <eos>"

If I have a paragraph, how to select sentences for encoder input and decoder input?

**Comment by JingqingZ** [comment]
1. Each letter actually represents one whole sentence. So the target "BD" has two sentences B and D, which are masked in the input.

2. How to select sentences for encoder input and decoder input? The sentence selection strategies (from a document) are described in Section 3.1 GSG in the paper.

**Comment by PingYu-iris** [comment]
I still don't understand. If our paragraph is "<s>ABCDE<e>":

From BART paper, their encoder input is "A_B_E", the decoder input is "<s>ABCD", the decoder output is "ABCDE". 

Let's take this for example, what is your encoder input and decoder input and output?

**Comment by JingqingZ** [comment]
As mentioned above,

Document: ABCDE

* BART: encoder input: A_C_E, decoder input/target: ABCDE
* PEGASUS: encoder input: A_C_E, decoder input/target: BD

Decoder input and decoder output (target) are mostly same, except shifted right (i.e. insert token <START> in decoder input).

**Comment by PingYu-iris** [comment]
Thanks
